### PR TITLE
chore: check for invalid input to factorial

### PR DESCRIPTION
**Pull request by abdullyahuza**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [ ] Add an algorithm?
- [x] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1229"></a>

### CODE REVIEW

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Maybe add a check for negative numbers while you are at it?

File: Recursive/Factorial.js [hunk 1]
```diff
@@ -9,6 +9,14 @@
  */
 
 const factorial = (n) => {
+  if (!(typeof n === 'number')) {
```
Comment (appgurueu) [inline Recursive/Factorial.js:12]: Why not use `!==` here?

File: Recursive/Factorial.js [hunk 2]
```diff
@@ -17,6 +17,10 @@ const factorial = (n) => {
     throw new RangeError('Not a Whole Number')
   }
 
+  if (n < 0) {
+    return undefined
```
Comment (appgurueu) [inline Recursive/Factorial.js:21]: Why not throw an error?

Comment (abdullyahuza) [reply Recursive/Factorial.js]: ok

**Review by appgurueu — APPROVED** [review]

**Comment by abdullyahuza** [comment]
All done.

File: Recursive/Factorial.js [hunk 3]
```diff
@@ -9,6 +9,18 @@
  */
 
 const factorial = (n) => {
+  if (typeof n !== 'number') {
+    throw new TypeError('Not a Number')
+  }
+
+  if ((n % 1) !== 0) {
```
Comment (raklaptudirm) [inline Recursive/Factorial.js:16]: Maybe use `Number.IsInteger(n)` here? Same functionality but more readable imo.

Comment (abdullyahuza) [reply Recursive/Factorial.js]: Done.
But It turns out that, checking `typeof n !== 'number'` is not necessary when `Number.isInteger(n)` is used.

Comment (raklaptudirm) [reply Recursive/Factorial.js]: That's true too, so simpler code.

**Comment by raklaptudirm** [comment]
You don't seem to have removed the `typeof` check, so do that.

File: Recursive/Factorial.js (hunk 3 above)
Comment (abdullyahuza) [reply Recursive/Factorial.js]: Okay.

**Review by raklaptudirm — APPROVED** [review]

**Review by appgurueu — APPROVED** [review]

**Pull request merged**